cmd/cert-checker/main.go

```diff
@@ -33,6 +34,26 @@ const (
 	expectedValidityPeriod = time.Hour * 24 * 90
 )
 
+// For defense-in-depth in addition to using the PA & its hostnamePolicy to
+// check domain names we also perform a check against the regex's from the
+// forbiddenDomains array
+var forbiddenDomainPatterns = []*regexp.Regexp{
+	regexp.MustCompile(`^\s*$`),
+	regexp.MustCompile(`\.mil$`),
+	regexp.MustCompile(`\.local$`),
+	regexp.MustCompile(`^localhost$`),
+	regexp.MustCompile(`\.localhost$`),
+}
+
+func isForbiddenDomain(name string) (bool, string) {
+	for _, r := range forbiddenDomainPatterns {
+		if matches := r.FindAllStringSubmatch(name, -1); len(matches) > 0 {
```

Comment:
I think it's fine as-is. Not too complex, and allows some sophistication in our checks.